lib/neo4j/rspec/matchers/properties.rb

```diff
@@ -31,14 +31,10 @@ module Properties
         matcher :define_property do |name, type = nil|
           match do |model|
             name = name.to_s
-            return false unless model.class.attributes.key?(name)
-
-            if type
-              nesting = Neo4j::RSpec::Compat.current.property_nesting
-              [nesting, Module].any? { |md| md.constants.any? { |c| type == c } }
-            else
-              true
-            end
+            expected_type = -> { type == :Boolean ? Neo4j::Shared::Boolean : Object.const_get(type.to_s) }
```

Comment:
@cheerfulstoic good on ya! For some reason I assumed that doing it in `Neo4j::Shared` context would assume everything was rooted in that (i.e. '`String'` would try to use `'Neo4j::Shared::String'`) 👍